File: src/python-pdk/hello_plugin.py
"""
Hello World Plugin for Extism using Python

This is a simple example of an Extism plugin written in Python.
"""

from extism_pdk import host

def hello():
    """
    A simple hello function that returns a greeting
    
    Returns:
        int: 0 for success, non-zero for error
    """
    # Get input from host
    input_text = host.get_input_string()
    if not input_text:
        input_text = "World"
    
    # Log debug message
    host.log_debug(f"Hello plugin called with input: {input_text}")
    
    # Create greeting
    greeting = f"Hello, {input_text}! This is an Extism plugin written in Python."
    
    # Set output
    host.set_output_string(greeting)
    
    # Return success
    return 0 

File: src/go-pdk/hello_plugin.go
package main

import (
	"./extism_pdk"
)

//export hello
func hello() int32 {
	// Create a host interface
	host := extism_pdk.CreateHost()

	// Get input from host
	input := host.GetInputString()
	if input == "" {
		input = "World"
	}

	// Log debug message
	host.LogDebug("Hello plugin called with input: " + input)

	// Create greeting
	greeting := "Hello, " + input + "! This is an Extism plugin written in Go."

	// Set output
	err := host.SetOutputString(greeting)
	if err != nil {
		host.LogError("Failed to set output: " + err.Error())
		host.SetError("Failed to set output")
		return 1
	}

	// Return success
	return 0
}

// This function is required for Go plugins
func main() {}
